assets/js/blocks-registry/payment-methods/registry.js

```diff
@@ -67,6 +67,33 @@ export const registerExpressPaymentMethod = ( options ) => {
 	}
 };
 
+/**
+ * Registers a callback for a specific payment method to determine if it can make payments
+ *
+ * @param {string} paymentMethodName A unique string to identify the payment method client side.
+ * @param {function():any } callBack Callback defined by extensions to determine if the payment method is supported.
+ */
+export const registerPaymentMethodExtensionCallback = (
```

Comment:
I am wondering what you think of changing the API to bring it more in line with the filters API. 

The API over there is like so: https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/131cb22b9c84574a88b6a4e411fabade5d1ed53c/packages/checkout/registry/index.ts#L34-L37

I was thinking we could do something similar here and change this to look like:

```js
export const registerPaymentMethodExtensionCallbacks = (
	namespace: string,
	paymentMethods: Record< string, PaymentMethodExtensionCallback >,
) 
```

Extensions would then use it like this:

```js
registerPaymentMethodExtensionCallbacks( 'test-extension'
	{
		stripe: ( arg ) => { /* code */ },
		paypal: ( arg ) => { /* code */ },
		cheque: ( arg ) => { /* code */ }
	}
);
```

Reply:
I was about to write the exact same comment. It's good for error reporting since we can show what extension caused the error. There is a negative tradeoff in that we would have to get the callbacks out of the namespaces in the first place, but this is only minor, and is outweighed by the positives.

Reply:
@opr I have added the changes and re-requested review.